.github/workflows/upload_app_store.yml

```diff
@@ -0,0 +1,66 @@
+name: Upload IPA to App Store
+on: workflow_dispatch
+jobs:
+  check_branch:
+    name: "Check branch ('release*' or 'main')."
+    runs-on: macos-latest
+    steps:
+      - id: early_exit
+        if: ${{ github.ref != 'refs/heads/main' && startsWith(github.ref, 'refs/heads/release') != true }}
+        name: Early exit failure if not main or release branch.
+        run: exit 1
+  build_test_upload:
+    name: Build, test, and upload to App Store
+    needs: check_branch
+    runs-on: macos-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+            node-version: 16
+            cache: "npm"
+            cache-dependency-path: "**/package-lock.json"
+      - name: Build core and features
+        run: ./build.js
+        env:
+            POLYPOD_POLYPEDIA_REPORT_URL: ${{ secrets.POLYPOD_POLYPEDIA_REPORT_URL }}
+            POLYPOD_POLYPEDIA_REPORT_AUTHORIZATION: ${{ secrets.POLYPOD_POLYPEDIA_REPORT_AUTHORIZATION }}
+      - name: Lint core and features
```

Comment:
You probably don't need this step at this point.

Reply:
I have added `Lint core and features` and `Test core and features`, as those are present in [upload to google play workflow](https://github.com/polypoly-eu/polyPod/blob/main/.github/workflows/upload_google_play.yml). I can remove  them if not needed.

Reply:
It's probably copypasta there too... At the point of uploading it, I don't think you really need to lint the code. Not that you can do anything if it does not pass...

Reply:
Well I was thinking of having this workflow to do any necessary sanity checks, at least `Test core and features` seems reasonable, though I am not sure if we should block the app upload if some of the core and features test fail.

I will cleanup those, and have only iOS tests as a prerequisite to uploading the app to the App Store.

Reply:
I have removed the additional lines, that are not very relevant for publishing the iOS App.